README.md

```diff
@@ -1,39 +1,362 @@
 
-The following is the template for the final README.md file:
+## Getting Started
 
----
+1. Add the `ReviewService` package to your project.
 
-# Project Title
+2. Create an instance of `ReviewService`. We'll cover dependency injection in details later on in this documentation.
+   ```dart
+   var reviewConditionsBuilder =
+      ReviewConditionsBuilder.empty().minimumPrimaryActionsCompleted(1);
 
-{Project tag line}
+   var reviewPrompter = LoggingReviewPrompter();
+   var reviewSettingsSource = MemoryReviewSettingsSource<ReviewSettings>();
 
-{Small description of the purpose of the project}
+   var reviewService = ReviewService<ReviewSettings>(
+      logger: Logger(),
+      reviewPrompter: reviewPrompter,
+      reviewSettingsSource: reviewSettingsSource,
+      reviewConditionsBuilder: reviewConditionsBuilder);
 
-[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](LICENSE)
+   ```
 
-## Getting Started
+3. Use the service.
+   - Update the review settings based on application events.
+      ```dart
+      late ReviewService<ReviewSettings> _reviewService;
+
+      Future<void> doPrimaryAction() async {
+         // Do Primary Action.
+
+         // Track this action.
+         await _reviewService.trackPrimaryActionCompleted();
+      }
+
+      ```
+   - Use the service to request review.
+      ```dart
+      late ReviewService<ReviewSettings> _reviewService;
+
+      Future<void> onCompletedImportantFlow() async {
+      // Do Meaningful Task.
+
+      // Check if all conditions are satisfied and prompt for review if they are.
+      await _reviewService.tryRequestReview();
+      }
+      ```
+
+## Next Steps
+
+### Persisting Review Settings
+
+`MemoryReviewSettingsSource` is great for automated testing but should not be the implementation of choice for real use-cases. Instead, you should create your own implementation that persists data on the device (so that review settings don't reset when you kill the app).
+
+```dart
+/// Storage implementation of <see cref="ReviewSettingsSource{TReviewSettings}"/>.
+sealed class StorageReviewSettingsSource<ReviewSettings> extends ReviewSettings
+    implements ReviewSettingsSource<ReviewSettings> {
+  @override
+  Future<ReviewSettings> read() {
+    // TODO: Return stored review settings.
+    return Future.value(ReviewSettings());
+  }
+
+  @override
+  Future<void> write(ReviewSettings reviewSettings) {
+    // TODO: Update stored review settings.
+    return Future.value();
+  }
+}
+```
+
+### Using Dependency Injection
+
+Here is a simple code that does dependency injection using `get_it`.
+
+```dart
+   GetIt getIt = GetIt.instance;
+
+   // Register the ReviewPrompter implementation with GetIt
+   getIt.registerSingleton<ReviewPrompter>(ReviewPrompter());
+```
+
+### Adding your CustomReviewSettings
```

Comment:
I wonder whether we should move this section in the Features section because it might not be a true next steps (because the default `ReviewSettings` type is designed to be usable directly).

Reply:
I moved it to features and renamed the section to 'Add Custom Tracking Data' which follows the previous section 'Add Custom Conditions'